doc/design/009-efficient-connections/README.md

```diff
@@ -0,0 +1,44 @@
+# Connections optimization design
+
+## Summary
+
+This design proposes enhancements of pool connection logic in libindy for
+reducing of pool load and better pool DDoS protection.
+
+## Motivation
+
+For current moment ```indy_pool_open``` endpoint performs CatchUp process and after this
+creates zmq sockets for each pool node and keep sockets connected until calling
+of ```indy_pool_close``` endpoint.
+
+This behavior causes the most of clients connected for the most of the time. As result
+we have obvious problem as each node can open only limited amount of sockets for
+the same time. Only first n clients can connect because n+1 connection will just
+cause error about limit of opened file descriptors.
+
+Additionally it makes easy to perform DDoS attack by justs calling ```indy_pool_open```
+in a cycle.
+
+Note that problem is complex and requires corresponded solution in Node codebase and
+pool network infrastructure, but this proposal is focused on libindy (client) side.
+
+## Proposed changes
+
+The main idea of proposal is force libindy to close sockets as soon as possible, but
+still provide very limited keep-alive ability to avoid unnecessary CurveCP and TCP
+handshakes:
+
+1. Change ```indy_pool_open``` endpoint behavior. Instead keep sockets connected it will only perform CatchUp
```

Comment:
Also the results of catch-up need to be persisted, so we don't do catch-up from scratch on every `indy_pool_open`

Reply:
I added this info